0002/README.md

```diff
@@ -0,0 +1,164 @@
+---
+id: 0002
+title: Release and SemVer
+status: ideation
+authors: Gianluca Arbezzano <gianarb92@gmail.com>
+---
+
+## Summary
+
+This proposal defines the outcome and the format we expect for each project
+and from Tinkerbell from a release point of view.
+
+## Goals and not Goals
+
+Goals are:
+
+1. What is a release
+2. A common format we expect from projects in order to be able to consume a
+   release from Tinkerbell
+3. Quality assertion.
+
+Not-Goals:
+
+1. How each project should be released
+2. Define who own releases in practice
+3. Distribution via `apt`, `apk` and so on.
+
+## Content
+
+A release is a tested and stable point in time snapshot from a codebase.
```

Comment:
To keep to the scope of this I think just making it, "A release is a tested, documented, and stable point in time snapshot from a codebase." should cover versioning the docs.

Reply:
Solved

Reply:
Should this also call out what types of artifacts we expect for a release?

For example:
- archive of the source code at the time of release
- binary build artifact(s) for a given release
- container image(s) for a given release
- OS specific packages for a given release
- etc

Reply:
I think this proposal should stay out of what each project exposes. The majority of the project is in Go, Osie is an OS distribution and it follows different rules. As I tried to explain but I will reinforce this is a generic high-level way approach to releasing software that every sub-project has to follow and apply.

I am sure we will have a follow-up issue/proposal or who knows about what Tinkerbell as a project will support in terms of OS, architectures, distribution, and so on and by consequence, the underline projects will have to get inline to those requirements.

Reply:
Osie results in a binary (executable at bootup, which tftp users would consume), does it not?
In Tinkerbell, is osie ever consumed as a container (maybe as an OCI filesystem / glorified-tarball)?

Reply:
It is a tarball and maybe it will become a set of tarball who knows right now. It can be confused as a docker image I am sure but don't know if we do it

Reply:
I think @DailyAlice's comment is not yet reflected here:
> A release is a tested, *documented*, and stable point in time snapshot from a codebase.

(It is reflected clearly below, https://github.com/tinkerbell/proposals/pull/5/files#diff-5ae2d202d9cad195081db97be153eaa9R139-R142)